Sign off previous commit 7abc6a8

Vicbi opened the pull request:
# *Sign off previous commit 7abc6a8*

## :recycle: Current situation & Problem
This PR aims to sign off a previous commit (7abc6a8) which was not signed at the time of submission. This is required for merging PR #8 .


## :gear: Release Notes 
Signed off previous commit 7abc6a8.


## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

PSchmiedmayer commented on README.md:
```diff
@@ -4,6 +4,8 @@ This source file is part of the Stanford Spezi open-source project.
 
 SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
 
+SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+
```
This looks like a text duplicate.

Vicbi replied:
Removed 👍

PSchmiedmayer reviewed (approved):
Thank you for improving the testing setup; great job!

Vicbi commented:
# Implement CI/Testing Setup and Update README Requirements

## ♻️ Current Situation & Problem

This PR addresses two open issues:

Issue #5 : Lack of CI and testing setup leading to build-breaking changes.
Issue #6 : Discrepancies between GitHub Actions requirements and README documentation.

This PR aims to enhance our project's development workflow by introducing a CI/Testing setup. Additionally, it updates the README to reflect new requirements for contributions. A crucial aspect of this PR is to address a specific concern from a previous phase of our project - a commit (`7abc6a8`) that was submitted without a signature, which is now required for merging PR #8.

## ⚙️ Release Notes

- Introduced a CI/testing setup using GitHub Actions, Firebase Emulator, and nbconvert for executing notebooks within CI, ensuring full integration tests without accessing production Firebase projects.

- Updated README to include all necessary installation requirements matching the GitHub Actions setup.

- Signed off previous commit `7abc6a8`.

### Examples:

- For setting up the Firebase Emulator and executing notebooks in CI:
```bash
firebase emulators:exec --import=./sample_data "jupyter nbconvert --to pdf --execute SpeziDataPipelineTemplate.ipynb"
```

- For including missing dependencies:
```bash
pip install pandas numpy matplotlib firebase-admin requests ipywidgets pytz
```

## 📚 Documentation
- Enhanced in-line documentation within the GitHub Actions workflow files to explain the purpose and function of each step in the CI process.

- Added instructions in the README to guide through the installation of project dependencies including requests, ipywidgets, and pytz, mirroring the setup in Google Colab.

## 📝 Code of Conduct & Contributing Guidelines

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):

I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).
